pkg/sql/conn_executor_exec.go

```diff
@@ -869,6 +873,38 @@ func (ex *connExecutor) dispatchToExecutionEngine(
 	return nil
 }
 
+// sampleLogicalPlanDescription returns a serialized representation of a statement's logical plan.
+// The returned ExplainTreePlanNode will be nil if plan should not be sampled.
+func (ex *connExecutor) sampleLogicalPlanDescription(
+	stmt Statement, useDistSQL bool, optimizerUsed bool, planner *planner,
+) *roachpb.ExplainTreePlanNode {
+	if !sampleLogicalPlans.Get(&ex.appStats.st.SV) {
+		return nil
+	}
+
+	if ex.saveLogicalPlanDescription(stmt, useDistSQL, optimizerUsed) {
+		return planToTree(context.Background(), planner.curPlan)
+	}
+	return nil
+}
+
+// saveLogicalPlanDescription returns if we should save this as a sample logical plan
+// for its corresponding fingerprint. We use `saveFingerprintPlanOnceEvery`
+// to assess how frequently to sample logical plans.
+func (ex *connExecutor) saveLogicalPlanDescription(
+	stmt Statement, useDistSQL bool, optimizerUsed bool,
+) bool {
+	stats := ex.appStats.getStatsForStmt(stmt, useDistSQL, optimizerUsed, nil, false /* createIfNonexistent */)
```

Comment:
yeah basically the call is not done in the right place. Since we're recording post-execution, this should be done at the end after the error status is known.